### PR TITLE
Fix emitting hidden warnings

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-devpack-dotnet/issues/751

### CODE REVIEW

File: src/Neo.Compiler.CSharp/CompilationContext.cs [hunk 1]
```diff
@@ -57,7 +57,7 @@ public class CompilationContext
         private readonly Dictionary<ITypeSymbol, byte> vtables = new(SymbolEqualityComparer.Default);
         private byte[]? script;
 
-        public bool Success => diagnostics.All(p => p.Severity != DiagnosticSeverity.Error);
+        public bool Success => !diagnostics.Any(p => p.Severity == DiagnosticSeverity.Error);
```
Comment (vncoelho) [inline src/Neo.Compiler.CSharp/CompilationContext.cs:60]: this is a good change

File: src/Neo.Compiler.CSharp/CompilationContext.cs [hunk 2]
```diff
@@ -124,7 +124,7 @@ private void Compile()
             foreach (SyntaxTree tree in compilation.SyntaxTrees)
             {
                 SemanticModel model = compilation.GetSemanticModel(tree);
-                diagnostics.AddRange(model.GetDiagnostics());
+                diagnostics.AddRange(model.GetDiagnostics().Where(u => u.Severity != DiagnosticSeverity.Hidden));
```
Comment (vncoelho) [inline src/Neo.Compiler.CSharp/CompilationContext.cs:127]: what this Hidden means?

Comment (shargon) [reply src/Neo.Compiler.CSharp/CompilationContext.cs:127]: Something that we don't need to show to the user :)

Comment (vncoelho) [reply src/Neo.Compiler.CSharp/CompilationContext.cs:127]: "These are typically used to provide suggestions or insights that could improve the code's quality, readability, or performance but are not critical errors or warnings that must be addressed for the code to run correctly."

In this case we are not talking about users, @shargon 
Devs are using these tools

Comment (vncoelho) [reply src/Neo.Compiler.CSharp/CompilationContext.cs:127]: In think the success change is good

Comment (shargon) [reply src/Neo.Compiler.CSharp/CompilationContext.cs:127]: > "These are typically used to provide suggestions or insights that could improve the code's quality, readability, or performance but are not critical errors or warnings that must be addressed for the code to run correctly."
> 
> In this case we are not talking about users, @shargon Devs are using these tools

Yes, devs hahah, but we don't need to show them these notifications https://github.com/neo-project/neo-devpack-dotnet/issues/751

**Review by vncoelho — APPROVED** [review]

File: src/Neo.Compiler.CSharp/CompilationContext.cs (hunk 1 above)
Comment (Jim8y) [inline src/Neo.Compiler.CSharp/CompilationContext.cs:60]: Hey, why this change? aren't them exactly the same?

Comment (shargon) [reply src/Neo.Compiler.CSharp/CompilationContext.cs:60]: yes xD
